### PR TITLE
OP(fusion_gru) error message enhancement. test=develop

### PR DESCRIPTION
**Pull request by zhaoyuchen2018**
c++ API enhancement

### CODE REVIEW

File: paddle/fluid/operators/fused/fusion_gru_op.cc [hunk 1]
```diff
@@ -24,51 +24,79 @@ namespace paddle {
...
+                        "WeightX size is:%d, WeightX dim is:[%s] ",
+                        wx_dims.size(), wx_dims));
   PADDLE_ENFORCE_EQ(wx_dims[0], x_dims[1],
-                    "The first dimension of Input(WeightX) "
-                    "should be %d.",
-                    x_dims[1]);
+                    platform::errors::InvalidArgument(
+                        "The first dimension of Input(WeightX) "
+                        "should equal to second dimension of input x, but "
+                        "received WeightX dimsneion is:%d, x dimension is:%d",
```
Comment (cyj1986) [inline paddle/fluid/operators/fused/fusion_gru_op.cc]: dimension

File: paddle/fluid/operators/fused/fusion_gru_op.cc [hunk 2]
```diff
@@ -24,51 +24,79 @@ namespace paddle {
...
+                        "is:%d, frame size is:%d",
+                        wh_dims[1], frame_size));
 
   if (ctx->HasInput("H0")) {
     auto h0_dims = ctx->GetInputDim("H0");
     PADDLE_ENFORCE_EQ(h0_dims[1], frame_size,
-                      "The width of H0 must be equal to frame_size.");
+                      platform::errors::InvalidArgument(
+                          "The width of H0 must be equal to frame_size, but "
+                          "receiced H0 is:%d, frame size is:%d",
```
Comment (cyj1986) [inline paddle/fluid/operators/fused/fusion_gru_op.cc]: the width of H0 is %d

Comment (zhaoyuchen2018) [reply paddle/fluid/operators/fused/fusion_gru_op.cc]: done

File: paddle/fluid/operators/fused/fusion_gru_op.cc (hunk 1 above)
Comment (zhaoyuchen2018) [reply paddle/fluid/operators/fused/fusion_gru_op.cc]: done

**Review by cyj1986 — DISMISSED** [review]

File: paddle/fluid/operators/fused/fusion_gru_op.cc [hunk 3]
```diff
@@ -24,51 +24,79 @@ namespace paddle {
...
+                        "The rank of Input(WeightH) should be 2, but received "
+                        "WeightH dim size is:%d, WeightH dim is:[%s]",
+                        wh_dims.size(), wh_dims));
   PADDLE_ENFORCE_EQ(wh_dims[0], frame_size,
-                    "The first dimension of Input(WeightH) "
-                    "should be %d.",
-                    frame_size);
+                    platform::errors::InvalidArgument(
+                        "The first dimension of Input(WeightH) "
+                        "should equal to frame_size, but received WeightH is: "
```
Comment (cyj1986) [inline paddle/fluid/operators/fused/fusion_gru_op.cc]: 描述不准确 the first dimension of WeightH is 或者the shape of  weightH is %s,

File: paddle/fluid/operators/fused/fusion_gru_op.cc [hunk 4]
```diff
@@ -24,51 +24,79 @@ namespace paddle {
...
+                        "Input(X)'s rank must be 2, but received input dim "
+                        "size is:%d, input dim is:[%s]",
+                        x_dims.size(), x_dims));
 
   auto wx_dims = ctx->GetInputDim("WeightX");
   PADDLE_ENFORCE_EQ(wx_dims.size(), 2,
-                    "The rank of Input(WeightX) should be 2.");
+                    platform::errors::InvalidArgument(
+                        "The rank of Input(WeightX) should be 2, but received "
+                        "WeightX size is:%d, WeightX dim is:[%s] ",
```
Comment (cyj1986) [inline paddle/fluid/operators/fused/fusion_gru_op.cc]: WeightX dim size

File: paddle/fluid/operators/fused/fusion_gru_op.cc (hunk 3 above)
Comment (zhaoyuchen2018) [reply paddle/fluid/operators/fused/fusion_gru_op.cc]: done

File: paddle/fluid/operators/fused/fusion_gru_op.cc (hunk 4 above)
Comment (zhaoyuchen2018) [reply paddle/fluid/operators/fused/fusion_gru_op.cc]: done

**Review by cyj1986 — APPROVED** [review]
LGTM

**Pull request merged**